Add waterstone

Solestis opened the pull request:
Added new block, waterstone.

Is fully functional. Textures may be improved later.

EoD commented:
Risking to repeat myself: Those blocks are awesome!

But I have a few things you might need to fix before I am going to merge this pull request:
- replace your name string like `"Black waterstone"` with something like `StatCollector.translateToLocal("tile.chisel.waterstone.1.desc")`
- Remove unfinished code like Lines 18-23 in BlockWaterstone.java
- Remove code which will never be used like `setLightLevel(1.0F)` (water is not shiny imho)
- Clean up imports (no need for `net.minecraft.world.World` in the current state

You can do this by doing a `git rebase -i HEAD a19de2cbfdc386c322841c59f5837e12d01e906e` and replace the `pick` by an `edit`. Then you apply your changes, use `git add` to add the files back and do a `git rebase --continue`.
This followed by a `git push -f origin add-waterstone` and I will merge this pull request :)

Of course it would also be great if you add those not-yet-implemented features, maybe even in a 3rd commit.

Solestis committed:
Removed unused lava texture

Removed unused (and illegal) lava_flow.png

Solestis committed:
Added Waterstone

Added:
New class for Waterstone, so particle effects may be implemented later.
Code to add Waterstone.
Crafting recipe for Waterstone.
Modified lavastone textures for Waterstone and any other fluid blocks in
the future.
English language entry for Waterstone.

EoD commented:
Much better, thanks!

![waterstone](https://cloud.githubusercontent.com/assets/293499/4446826/b37a73a8-4805-11e4-843d-8f7d0bf758fb.png)
